File: src/js/sdk/test/urlUtils.test.ts
import { isExternalCallAllowed } from '../src/urlUtils';

describe('isExternalCallAllowed', () => {
    it('allows all when disableExternalCalls is false', () => {
        expect(isExternalCallAllowed('https://api.lukuid.com', false)).toBe(true);
        expect(isExternalCallAllowed('https://custom.endpoint.com', false)).toBe(true);
    });

    it('blocks lukuid.com when disableExternalCalls is true', () => {
        expect(isExternalCallAllowed('https://api.lukuid.com', true)).toBe(false);
        expect(isExternalCallAllowed('http://lukuid.com', true)).toBe(false);
        expect(isExternalCallAllowed('https://sub.api.lukuid.com/path', true)).toBe(false);
        expect(isExternalCallAllowed('https://LUKUID.COM', true)).toBe(false);
    });

    it('allows custom endpoints when disableExternalCalls is true', () => {
        expect(isExternalCallAllowed('https://custom.endpoint.com', true)).toBe(true);
        expect(isExternalCallAllowed('https://notlukuid.com', true)).toBe(true);
        expect(isExternalCallAllowed('http://localhost:8080', true)).toBe(true);
        expect(isExternalCallAllowed('https://lukuid.com.br', true)).toBe(true);
    });
});

File: src/android/lukuid-sdk/src/test/java/com/lukuid/sdk/internal/UrlUtilsTest.kt
package com.lukuid.sdk.internal

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class UrlUtilsTest {
    @Test
    fun testIsExternalCallAllowed() {
        // Disabled = false
        assertTrue(UrlUtils.isExternalCallAllowed("https://api.lukuid.com", false))
        assertTrue(UrlUtils.isExternalCallAllowed("https://custom.endpoint.com", false))

        // Disabled = true, block lukuid.com
        assertFalse(UrlUtils.isExternalCallAllowed("https://api.lukuid.com", true))
        assertFalse(UrlUtils.isExternalCallAllowed("http://lukuid.com", true))
        assertFalse(UrlUtils.isExternalCallAllowed("https://sub.api.lukuid.com/path", true))
        assertFalse(UrlUtils.isExternalCallAllowed("https://LUKUID.COM", true))

        // Disabled = true, allow custom endpoints
        assertTrue(UrlUtils.isExternalCallAllowed("https://custom.endpoint.com", true))
        assertTrue(UrlUtils.isExternalCallAllowed("https://notlukuid.com", true))
        assertTrue(UrlUtils.isExternalCallAllowed("http://localhost:8080", true))
        assertTrue(UrlUtils.isExternalCallAllowed("https://lukuid.com.br", true))
    }
}
